source/codegen/templates/proto_helpers.mako

```diff
@@ -114,21 +135,36 @@ message ${common_helpers.snake_to_pascal(function)}Response {
 % for parameter in output_parameters:
 <%
   parameter_name = common_helpers.camel_to_snake(parameter["name"])
-  grpc_field_number = proto_helpers.generate_parameter_field_number(parameter, used_indexes)
+  is_array = common_helpers.is_array(parameter["type"])
   if "grpc_type" in parameter:
     parameter_type = parameter["grpc_type"]
   else:
-    parameter_type = proto_helpers.determine_function_parameter_type(parameter, service_class_prefix)
+    parameter_type = proto_helpers.get_grpc_type_from_ivi(parameter["type"], is_array, service_class_prefix)
 %>\
 %   if common_helpers.is_enum(parameter):
 <%
-  is_array = common_helpers.is_array(parameter["type"])
-  non_enum_type = proto_helpers.get_grpc_type_from_ivi(parameter["type"], is_array, service_class_prefix)
+  enum_parameter_type = None
```

Comment:
This really should be shared somehow with the input case. Or at least have a bunch of shared helpers.